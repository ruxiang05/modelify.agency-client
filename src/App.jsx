import React from 'react';
import Header from './components/Header';
import Main from './components/Main';
import { UserProvider } from './contexts/userContext';
import './styles/app.scss';
import './styles/common.scss';
import { getCurrentUser, logOut } from './auth';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      user: getCurrentUser(),
    };

    this.updateUser = this.updateUser.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
  }

  updateUser() {
    this.setState({ user: getCurrentUser() });
  }

  handleLogout() {
    logOut();
    this.setState({ user: null });
  }

  render() {
    const { user } = this.state;
    return (
      <UserProvider value={{ user, updateUser: this.updateUser, handleLogout: this.handleLogout }}>
        <div className="app">
          {user && <Header user={user} />}
          <Main user={user} />
        </div>
      </UserProvider>
    );
  }
}

export default App;
