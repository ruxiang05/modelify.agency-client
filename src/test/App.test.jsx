import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});

describe('App', () => {
  it('renders App', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
