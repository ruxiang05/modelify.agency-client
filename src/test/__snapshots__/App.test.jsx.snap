// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Context.Provider
        value={
          Object {
            "socket": undefined,
          }
        }
      >
        <div
          className="app"
        >
          <Main
            user={null}
          />
        </div>
      </Context.Provider>,
      "value": Object {
        "handleLogout": [Function],
        "updateUser": [Function],
        "user": null,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="app"
        >
          <Main
            user={null}
          />
        </div>,
        "value": Object {
          "socket": undefined,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <Main
              user={null}
            />,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "user": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {
            "socket": Object {},
          },
          "_currentValue2": Object {
            "socket": Object {},
          },
          "_threadCount": 0,
        },
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "updateUser": [Function],
          "user": Object {},
        },
        "_currentValue2": Object {
          "updateUser": [Function],
          "user": Object {},
        },
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Context.Provider
          value={
            Object {
              "socket": undefined,
            }
          }
        >
          <div
            className="app"
          >
            <Main
              user={null}
            />
          </div>
        </Context.Provider>,
        "value": Object {
          "handleLogout": [Function],
          "updateUser": [Function],
          "user": null,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="app"
          >
            <Main
              user={null}
            />
          </div>,
          "value": Object {
            "socket": undefined,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <Main
                user={null}
              />,
            ],
            "className": "app",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "user": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": Object {
          "$$typeof": Symbol(react.provider),
          "_context": Object {
            "$$typeof": Symbol(react.context),
            "Consumer": Object {
              "$$typeof": Symbol(react.context),
              "_calculateChangedBits": null,
              "_context": [Circular],
            },
            "Provider": [Circular],
            "_calculateChangedBits": null,
            "_currentRenderer": null,
            "_currentRenderer2": null,
            "_currentValue": Object {
              "socket": Object {},
            },
            "_currentValue2": Object {
              "socket": Object {},
            },
            "_threadCount": 0,
          },
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {
            "updateUser": [Function],
            "user": Object {},
          },
          "_currentValue2": Object {
            "updateUser": [Function],
            "user": Object {},
          },
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
